### ISSUE TITLE
ClaimsPrincipal input parameter is null (migrating netcore3.1 -> net5.0)

### ISSUE THREAD

**Issue by josmithua** [Azure/azure-functions-dotnet-worker#561]
I am migrating a Function app to target .NET 5, and now the `ClaimsPrincipal` instance from the input parameter is always `null`. Does the worker support the ClaimsPrincipal binding?

Thanks

```cs
namespace Blah {
    public static class MyFunction {
        [Function("MyFunction")]
        public static async Task<HttpResponseData> Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "test")] HttpRequestData req,
            ClaimsPrincipal principal)
        {
            if (principal == null)
            {
                // always hits here
            }
        }
    }
}
```

**Comment by david-peden-q2** [comment]
@josmithua see https://github.com/Azure/azure-functions-dotnet-worker/issues/436 and https://github.com/Azure/azure-functions-dotnet-worker/issues/574 for related discussion.

**Comment by josmithua** [comment]
@david-peden-q2 I'm still not sure how to access the ClaimsPrincipal from the function..

**Comment by david-peden-q2** [comment]
@josmithua You can follow @dolphinspired's [post](https://github.com/Azure/azure-functions-dotnet-worker/issues/574#issuecomment-911938184) and replace `FunctionContext` with `ClaimsPrincipal`.

**Comment by josmithua** [comment]
Thanks @tranb3r! I can confirm this works, however when running in localhost, any method call on `req.Identities` will throw an error (ex: `req.Identities.Count()`). The `ClaimsPrincipal` constructor enumerates the identities list, so you have to wrap the constructor call in a `try catch`

**Comment by kshyju** [comment]
@josmithua The null reference exception is fixed in version [1.6.0 of the Microsoft.Azure.Functions.Worker](https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker/1.6.0). Please give that version a try.

**Comment by josmithua** [comment]
@kshyju Yep, seems this is fixed! 

I will close this issue. It would have been nice to keep this as an input parameter, but the one liner works fine.
